src/core/ext/filters/rbac/rbac_service_config_parser.cc

```diff
@@ -716,28 +735,65 @@ const JsonLoaderInterface* RbacConfig::RbacPolicy::Rules::Policy::JsonLoader(
   return loader;
 }
 
+//
+// RbacConfig::RbacPolicy::Rules::AuditLogger
+//
+
+const JsonLoaderInterface*
+RbacConfig::RbacPolicy::Rules::AuditLogger::JsonLoader(const JsonArgs&) {
+  // All fields handled in JsonPostLoad().
+  static const auto* loader = JsonObjectLoader<AuditLogger>().Finish();
+  return loader;
+}
+
+void RbacConfig::RbacPolicy::Rules::AuditLogger::JsonPostLoad(
+    const Json& json, const JsonArgs& args, ValidationErrors* errors) {
+  // Should have exactly one field as the logger name.
+  if (json.object().size() != 1) {
+    errors->AddError("audit logger should have exactly one field");
+    return;
+  }
+  name = json.object().begin()->first;
+  auto config_or =
+      LoadJsonObjectField<Json::Object>(json.object(), args, name, errors);
+  if (config_or.has_value()) {
+    config = std::move(config_or.value());
+  }
+}
+
 //
 // RbacConfig::RbacPolicy::Rules
 //
 
-Rbac RbacConfig::RbacPolicy::Rules::TakeAsRbac() {
+Rbac RbacConfig::RbacPolicy::Rules::TakeAsRbac(std::string name) {
   Rbac rbac;
-  // TODO(lwge): This is to fix msan failure for now. Add proper conversion once
-  // audit logging support is added.
-  rbac.audit_condition = Rbac::AuditCondition::kNone;
+  rbac.name = std::move(name);
   rbac.action = static_cast<Rbac::Action>(action);
+  rbac.audit_condition = static_cast<Rbac::AuditCondition>(audit_condition);
   for (auto& p : policies) {
     rbac.policies.emplace(p.first, p.second.TakeAsRbacPolicy());
   }
+  for (auto& logger : audit_loggers) {
+    auto config_or = AuditLoggerRegistry::ParseConfig(
+        logger.name, Json::FromObject(std::move(logger.config)));
+    // The config should have been validated when the service config is
```

Comment:
To be clear, what I meant was the logger configs within the service config were already good ones when they are generated by the xDS logger registry. But I fully agree it's beneficial to validate them here as well.

I changed the logic following your suggestion. PTAL.

Reply:
I agree that this doesn't make a difference in practice, since the only way to configure this filter is via xDS.  However, more generally, our current code is designed to pass the configs to the xDS HTTP filters via the service config, so that if we have cause to in the future, it won't be hard to allow the filter to also be configured in non-xDS cases.  In the case of this particular filter, it seems unlikely that we would ever want to configure it in a non-xDS case, since users would probably just use the SDK authz API instead, but I'd prefer to keep everything structured in a consistent way.

(It is also worth noting that the other languages don't do it this way; they actually create interceptors for the xDS HTTP filters somewhat dynamically for each request, so they don't have to pass the config down like this.  Currently, C-core can't do it that way, since our filters need to be instantiated when the channel stack is instantiated, which is before we choose the config for each request.  But once we're done converting our stack from the legacy filter API to the new promise-based filter API, we probably will be able to do something like the other languages.  And at that point, we'll have to have a cross-language conversation about whether or not that's what we want, since it would remove the potential to be able to configure these filters via service config if we wanted to do that at some point in the future.  But it certainly would remove a lot of code for encoding the config in JSON and then re-parsing it in the filter.)

Anyway, the changes you made here look good!